File: Practice/AllDomains/Tutorials/30DaysOfCode/Day112DArrays.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution 
{
	/*	author:@shivkrthakur  */
    static void Main(String[] args) 
	{
        int arrLength = 6;
        int[][] arr = new int[arrLength][];
        for(int arr_i = 0; arr_i < arrLength; arr_i++){
           string[] arr_temp = Console.ReadLine().Split(' ');
           arr[arr_i] = Array.ConvertAll(arr_temp,Int32.Parse);
        }
        
        int sum = int.MinValue;
        for(int j = 0; j < arrLength; j++)
        {
            int length = arr[j].Length;
            
            for(int k=0; k < length; k++)
            {
                if((j+2) < arrLength && (k+2) < length)
                {
                        int currentSum = arr[j][k] + arr[j][k + 1] + arr[j][k + 2] + arr[j + 1][k + 1] + arr[j + 2][k] + arr[j + 2][k + 1]+ arr[j + 2][k + 2];
                        if(currentSum > sum) sum = currentSum;
                }
                
            }
        }
        Console.WriteLine(sum);
	}
}

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day112DArrays.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var arr = [];
    for(arr_i = 0; arr_i < 6; arr_i++){
       arr[arr_i] = readLine().split(' ');
       arr[arr_i] = arr[arr_i].map(Number);
    }
    var max = Number.MIN_SAFE_INTEGER;
    for(var i = 0; i < arr.length - 2; i++){
      for(var j = 0; j < arr.length - 2; j++){
          var sum = 0;
          var flag = true;
          for(var row = 0; row < 3; row++){
            if(flag){
              for(var col = 0; col < 3; col++){
                sum += arr[i + row][j + col];
              }
            }
            else sum += arr[i + row][j + 1]
            flag = !flag;
          }
          max = Math.max(sum, max);
      }
    }
    console.log(max);
}
